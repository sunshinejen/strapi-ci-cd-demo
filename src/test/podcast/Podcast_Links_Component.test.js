import React from 'react'
import { render, screen } from '@testing-library/react';
import PodcastLinks  from '../../components/podcast/Podcast_links'

// smoke test
it("Renders PodcastLinks component without crashing", () => {
    render(<PodcastLinks soundCloud='soundcloud.url'
        spotify='spotify.url' podcasts='apple.url'
    />)
})

// Snapshot Test. Test will nofify dev if component is modified.
test("component matches snapshot", () => {
    const { asFragment } = render(<PodcastLinks />)
    expect(asFragment()).toMatchSnapshot()
})


test("PodcastLinks Unit Test", () => {
    // render component with href props.
    render(<PodcastLinks 
        soundCloud='soundcloud.url_unitTest'
        spotify='spotify.url_unitTest'
        podcasts='apple.url_unitTest'
    />)

    // Test for href link in Document.
    expect(screen.getByTestId('soundCloud-link').href).toBe("http://localhost/soundcloud.url_unitTest")
    expect(screen.getByTestId('spotify-link').href).toBe("http://localhost/spotify.url_unitTest")
    expect(screen.getByTestId('apple-link').href).toBe("http://localhost/apple.url_unitTest")
})

  