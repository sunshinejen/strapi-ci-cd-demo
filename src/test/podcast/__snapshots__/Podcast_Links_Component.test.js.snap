// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="podcast-links-buttons"
  >
    <a
      class="link-tag"
      data-testid="soundCloud-link"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="hover-text tooltiptext"
      >
         Soundcloud 
      </span>
      <svg
        aria-labelledby="svg-inline--fa-title-hMRU5LpcqfWM"
        class="svg-inline--fa fa-soundcloud fa-w-20 podbtn fa-2x"
        data-icon="soundcloud"
        data-placement="bottom"
        data-prefix="fab"
        data-toggle="tooltip"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-hMRU5LpcqfWM"
        >
          Soundcloud
        </title>
        <path
          d="M111.4 256.3l5.8 65-5.8 68.3c-.3 2.5-2.2 4.4-4.4 4.4s-4.2-1.9-4.2-4.4l-5.6-68.3 5.6-65c0-2.2 1.9-4.2 4.2-4.2 2.2 0 4.1 2 4.4 4.2zm21.4-45.6c-2.8 0-4.7 2.2-5 5l-5 105.6 5 68.3c.3 2.8 2.2 5 5 5 2.5 0 4.7-2.2 4.7-5l5.8-68.3-5.8-105.6c0-2.8-2.2-5-4.7-5zm25.5-24.1c-3.1 0-5.3 2.2-5.6 5.3l-4.4 130 4.4 67.8c.3 3.1 2.5 5.3 5.6 5.3 2.8 0 5.3-2.2 5.3-5.3l5.3-67.8-5.3-130c0-3.1-2.5-5.3-5.3-5.3zM7.2 283.2c-1.4 0-2.2 1.1-2.5 2.5L0 321.3l4.7 35c.3 1.4 1.1 2.5 2.5 2.5s2.2-1.1 2.5-2.5l5.6-35-5.6-35.6c-.3-1.4-1.1-2.5-2.5-2.5zm23.6-21.9c-1.4 0-2.5 1.1-2.5 2.5l-6.4 57.5 6.4 56.1c0 1.7 1.1 2.8 2.5 2.8s2.5-1.1 2.8-2.5l7.2-56.4-7.2-57.5c-.3-1.4-1.4-2.5-2.8-2.5zm25.3-11.4c-1.7 0-3.1 1.4-3.3 3.3L47 321.3l5.8 65.8c.3 1.7 1.7 3.1 3.3 3.1 1.7 0 3.1-1.4 3.1-3.1l6.9-65.8-6.9-68.1c0-1.9-1.4-3.3-3.1-3.3zm25.3-2.2c-1.9 0-3.6 1.4-3.6 3.6l-5.8 70 5.8 67.8c0 2.2 1.7 3.6 3.6 3.6s3.6-1.4 3.9-3.6l6.4-67.8-6.4-70c-.3-2.2-2-3.6-3.9-3.6zm241.4-110.9c-1.1-.8-2.8-1.4-4.2-1.4-2.2 0-4.2.8-5.6 1.9-1.9 1.7-3.1 4.2-3.3 6.7v.8l-3.3 176.7 1.7 32.5 1.7 31.7c.3 4.7 4.2 8.6 8.9 8.6s8.6-3.9 8.6-8.6l3.9-64.2-3.9-177.5c-.4-3-2-5.8-4.5-7.2zm-26.7 15.3c-1.4-.8-2.8-1.4-4.4-1.4s-3.1.6-4.4 1.4c-2.2 1.4-3.6 3.9-3.6 6.7l-.3 1.7-2.8 160.8s0 .3 3.1 65.6v.3c0 1.7.6 3.3 1.7 4.7 1.7 1.9 3.9 3.1 6.4 3.1 2.2 0 4.2-1.1 5.6-2.5 1.7-1.4 2.5-3.3 2.5-5.6l.3-6.7 3.1-58.6-3.3-162.8c-.3-2.8-1.7-5.3-3.9-6.7zm-111.4 22.5c-3.1 0-5.8 2.8-5.8 6.1l-4.4 140.6 4.4 67.2c.3 3.3 2.8 5.8 5.8 5.8 3.3 0 5.8-2.5 6.1-5.8l5-67.2-5-140.6c-.2-3.3-2.7-6.1-6.1-6.1zm376.7 62.8c-10.8 0-21.1 2.2-30.6 6.1-6.4-70.8-65.8-126.4-138.3-126.4-17.8 0-35 3.3-50.3 9.4-6.1 2.2-7.8 4.4-7.8 9.2v249.7c0 5 3.9 8.6 8.6 9.2h218.3c43.3 0 78.6-35 78.6-78.3.1-43.6-35.2-78.9-78.5-78.9zm-296.7-60.3c-4.2 0-7.5 3.3-7.8 7.8l-3.3 136.7 3.3 65.6c.3 4.2 3.6 7.5 7.8 7.5 4.2 0 7.5-3.3 7.5-7.5l3.9-65.6-3.9-136.7c-.3-4.5-3.3-7.8-7.5-7.8zm-53.6-7.8c-3.3 0-6.4 3.1-6.4 6.7l-3.9 145.3 3.9 66.9c.3 3.6 3.1 6.4 6.4 6.4 3.6 0 6.4-2.8 6.7-6.4l4.4-66.9-4.4-145.3c-.3-3.6-3.1-6.7-6.7-6.7zm26.7 3.4c-3.9 0-6.9 3.1-6.9 6.9L227 321.3l3.9 66.4c.3 3.9 3.1 6.9 6.9 6.9s6.9-3.1 6.9-6.9l4.2-66.4-4.2-141.7c0-3.9-3-6.9-6.9-6.9z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      class="link-tag"
      data-testid="spotify-link"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="hover-text tooltiptext"
      >
         Spotify 
      </span>
      <svg
        aria-labelledby="svg-inline--fa-title-fRa72RdGvvjd"
        class="svg-inline--fa fa-spotify fa-w-16 podbtn fa-2x"
        data-icon="spotify"
        data-placement="bottom"
        data-prefix="fab"
        data-toggle="tooltip"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-fRa72RdGvvjd"
        >
          Spotify
        </title>
        <path
          d="M248 8C111.1 8 0 119.1 0 256s111.1 248 248 248 248-111.1 248-248S384.9 8 248 8zm100.7 364.9c-4.2 0-6.8-1.3-10.7-3.6-62.4-37.6-135-39.2-206.7-24.5-3.9 1-9 2.6-11.9 2.6-9.7 0-15.8-7.7-15.8-15.8 0-10.3 6.1-15.2 13.6-16.8 81.9-18.1 165.6-16.5 237 26.2 6.1 3.9 9.7 7.4 9.7 16.5s-7.1 15.4-15.2 15.4zm26.9-65.6c-5.2 0-8.7-2.3-12.3-4.2-62.5-37-155.7-51.9-238.6-29.4-4.8 1.3-7.4 2.6-11.9 2.6-10.7 0-19.4-8.7-19.4-19.4s5.2-17.8 15.5-20.7c27.8-7.8 56.2-13.6 97.8-13.6 64.9 0 127.6 16.1 177 45.5 8.1 4.8 11.3 11 11.3 19.7-.1 10.8-8.5 19.5-19.4 19.5zm31-76.2c-5.2 0-8.4-1.3-12.9-3.9-71.2-42.5-198.5-52.7-280.9-29.7-3.6 1-8.1 2.6-12.9 2.6-13.2 0-23.3-10.3-23.3-23.6 0-13.6 8.4-21.3 17.4-23.9 35.2-10.3 74.6-15.2 117.5-15.2 73 0 149.5 15.2 205.4 47.8 7.8 4.5 12.9 10.7 12.9 22.6 0 13.6-11 23.3-23.2 23.3z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      class="link-tag"
      data-testid="apple-link"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="hover-text tooltiptext"
      >
         Apple 
      </span>
      <svg
        aria-labelledby="svg-inline--fa-title-BA18LgpDBaVP"
        class="svg-inline--fa fa-podcast fa-w-14 podbtn fa-2x"
        data-icon="podcast"
        data-placement="bottom"
        data-prefix="fas"
        data-toggle="tooltip"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-BA18LgpDBaVP"
        >
          Apple Podcasts
        </title>
        <path
          d="M267.429 488.563C262.286 507.573 242.858 512 224 512c-18.857 0-38.286-4.427-43.428-23.437C172.927 460.134 160 388.898 160 355.75c0-35.156 31.142-43.75 64-43.75s64 8.594 64 43.75c0 32.949-12.871 104.179-20.571 132.813zM156.867 288.554c-18.693-18.308-29.958-44.173-28.784-72.599 2.054-49.724 42.395-89.956 92.124-91.881C274.862 121.958 320 165.807 320 220c0 26.827-11.064 51.116-28.866 68.552-2.675 2.62-2.401 6.986.628 9.187 9.312 6.765 16.46 15.343 21.234 25.363 1.741 3.654 6.497 4.66 9.449 1.891 28.826-27.043 46.553-65.783 45.511-108.565-1.855-76.206-63.595-138.208-139.793-140.369C146.869 73.753 80 139.215 80 220c0 41.361 17.532 78.7 45.55 104.989 2.953 2.771 7.711 1.77 9.453-1.887 4.774-10.021 11.923-18.598 21.235-25.363 3.029-2.2 3.304-6.566.629-9.185zM224 0C100.204 0 0 100.185 0 224c0 89.992 52.602 165.647 125.739 201.408 4.333 2.118 9.267-1.544 8.535-6.31-2.382-15.512-4.342-30.946-5.406-44.339-.146-1.836-1.149-3.486-2.678-4.512-47.4-31.806-78.564-86.016-78.187-147.347.592-96.237 79.29-174.648 175.529-174.899C320.793 47.747 400 126.797 400 224c0 61.932-32.158 116.49-80.65 147.867-.999 14.037-3.069 30.588-5.624 47.23-.732 4.767 4.203 8.429 8.535 6.31C395.227 389.727 448 314.187 448 224 448 100.205 347.815 0 224 0zm0 160c-35.346 0-64 28.654-64 64s28.654 64 64 64 64-28.654 64-64-28.654-64-64-64z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
